Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/CalculateMonteCarloAbsorption.py

```diff
@@ -0,0 +1,794 @@
+                ss_monte_carlo_alg.execute()
+                acc_2 = ss_monte_carlo_alg.getProperty("OutputWorkspace").value
+
+                acc_ws = self._multiply(acc_1, acc_2)
+
+            for log_name, log_value in container_kwargs.items():
+                sample_log_names.append("container_" + log_name.lower())
+                sample_log_values.append(log_value)
+
+            acc_ws = self._convert_from_wavelength(acc_ws)
+
+            self._add_sample_log_multiple(acc_ws, sample_log_names, sample_log_values)
+
+            if not self.isChild():
+                mtd.addOrReplace(self._acc_ws_name, acc_ws)
+
+            self._output_ws = self._group_ws([ass_ws, acc_ws])
+        else:
+            self._output_ws = self._group_ws([ass_ws])
+
+        self.setProperty('CorrectionsWorkspace', self._output_ws)
+
+    def _setup(self):
+
+        # The beam properties and monte carlo properties are simply passed straight on to the
+        # SimpleShapeMonteCarloAbsorptionCorrection algorithm so they are being put into
+        # a dictionary for simplicity
+        sample_workspace = self.getProperty("SampleWorkspace").value
+        container_workspace = self.getProperty("ContainerWorkspace").value
+        sample_is_group = isinstance(sample_workspace, WorkspaceGroup)
+        container_is_group = isinstance(container_workspace, WorkspaceGroup)
+
+        if container_workspace and sample_is_group != container_is_group:
+            sample_type = "WorkspaceGroup" if sample_is_group else "MatrixWorkspace"
+            container_type = "WorkspaceGroup" if container_is_group else "MatrixWorkspace"
+            raise RuntimeError("Mismatch between SampleWorkspace (" + sample_type + ") and"
+                               " ContainerWorkspace (" + container_type + ").")
+        elif sample_is_group and (container_is_group or not container_workspace):
+
+            if container_workspace and len(sample_workspace) != len(container_workspace):
+                raise RuntimeError("SampleWorkspace group and ContainerWorkspace group do not"
+                                   " have the same number of workspaces.")
+            return
+
+        self._general_kwargs = {'BeamHeight': self.getProperty('BeamHeight').value,
+                                'BeamWidth': self.getProperty('BeamWidth').value,
+                                'NumberOfWavelengthPoints': self.getProperty('NumberOfWavelengthPoints').value,
+                                'EventsPerPoint': self.getProperty('EventsPerPoint').value,
+                                'Interpolation': self.getProperty('Interpolation').value}
+
+        self._sample_ws = sample_workspace
+        self._container_ws = container_workspace
+        self._shape = self.getProperty('Shape').value
+        self._height = self.getProperty('Height').value
+
+        self._sample_unit = self._sample_ws.getAxis(0).getUnit().unitID()
```

Comment:
Now when you tried to get it working for groups, this causes a problem, because `self._sample_ws` does not have getAxis for groups. I get `None type has no attribute getAxis()` at validation step.
I think this should be wrapped in a check, we are dealing with group workspaces, we should take the unit probably from the first workspace in the group.

Reply:
This step should never actually be reached for the group workspace itself -> it will later be done for each workspace in turn.

Reply:
Could you give me the steps you took to produce this issue?